airgen/src/lib.rs

```diff
@@ -26,13 +26,14 @@ const MAIN_FUNCTION: &str = "main";
 pub fn compile(input: AnalysisASMFile) -> PILGraph {
     let main_location = Location::main();
 
-    let non_std_machines = input
+    let neither_std_nor_rom_machines = input
         .machines()
         .filter(|(k, _)| k.parts().next() != Some("std"))
+        .filter(|(k, _)| !k.parts().last().unwrap().contains("ROM"))
```

Comment:
Who supplies ROM and what makes a machine a ROM machine?

I still don't understand the purpose of this PR but shouldn't the detection based on semantics instead of the name?

Reply:
> Who supplies ROM and what makes a machine a ROM machine?

The ROM machine is generated internally from the machine's program. I agree the detection can be better here.

Reply:
I tried to come up with something more clever but I think this is fine. This would only fail if the user's main machine ends with "ROM".. I don't think it makes a big difference to base it on the inside of the machine, for example having a single operation named `get_line`.